llvm/lib/Transforms/Utils/CodeLayout.cpp

```diff
@@ -1152,6 +1153,7 @@ class CDSortImpl {
       MergeGainT BestGain = BestEdge->getMergeGain();
       mergeChains(BestSrcChain, BestDstChain, BestGain.mergeOffset(),
                   BestGain.mergeType());
+      --NumHotChains;
```

Comment:
Seems like the list, HotChains, is not used in the algorithm. Can we get rid of it altogether?

Instead of `for (ChainT *ChainPred : HotChains) {...` we'd have something like
```
for (NodeT &Node : AllNodes) {
  if (Node.ExecutionCount > 0) {
    ChainT *ChainPred = Node.CurChain;
    ...
  }
}
```

Reply:
Thanks for the suggestion. I'll optimize out `HotChains` and rename `NumHotChains` to `NumActiveChains`.